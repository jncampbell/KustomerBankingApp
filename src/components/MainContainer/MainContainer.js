import React from 'react';
import PropTypes from 'prop-types';
import BankForm from '../../components/BankForm';
import './styles/main-container.scss';

class MainContainer extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className='bank'>
        <div className='bank-section bank__balance bank-balance'>
          <div className='bank-balance__text'>Your current balance is: </div>
          <div className='bank-balance__amount'>${this.props.balance.toFixed(2)}</div>
        </div>
        <div className='bank-section'>
          <div className='bank-section__cta'>Deposit Funds</div>
          <div className='bank-section__form'>
              <BankForm processHandler={this.props.processDeposit} type='deposit'/>
          </div>
        </div>
        <div className='bank-section'>
          <div className='bank-section__cta'>Withdraw Funds</div>
          <div className='bank-section__form'>
            <BankForm processHandler={this.props.processWithdrawal} type='withdrawal' />
          </div>
        </div>
        <div className='bank-section'>
           <div className='bank-section__cta'>Show Recent Transactions</div>
           <div className='bank-recent-transactions'>
             {/* Not happy with the following setup. Would refactor */}
            <div className="divTable" >
              <div className="divTableBody">
                <div className="divTableRow">
                  <div className='divTableRow'>
                    <div className='divTableCell'>Date</div>
                    <div className='divTableCell'>Amount</div>
                    <div className='divTableCell'>Balance</div>
                  </div>
                  {this.props.recentTransactions.map((transaction, i) => {
                    return (
                    <div className='divTableRow' key={i}>
                      <div className='divTableCell'>{transaction.timestamp}</div>
                      <div className='divTableCell'>{transaction.type === 'withdrawal' ? "-" : ' '} {transaction.amount.toFixed(2)}</div>
                      <div className='divTableCell'>{transaction.balance.toFixed(2)}</div>
                    </div>
                   );
                  })}
                </div>
              </div>
            </div>

           </div>
      </div>
    </div>
    );
  }
}

MainContainer.propTypes = {
  balance: PropTypes.number.isRequired,
  processDeposit: PropTypes.func.isRequired,
  processWithdrawal: PropTypes.func.isRequired,
  recentTransactions: PropTypes.array
}

export default MainContainer;
