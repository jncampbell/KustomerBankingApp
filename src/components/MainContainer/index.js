import {connect} from 'react-redux';
import {deposit, withdrawal} from '../../redux/actions';
import MainContainer from './MainContainer';

const mapStateToProps = (state) => ({
  balance: state.balance,
  recentTransactions: JSON.parse(JSON.stringify(state.recentTransactions))
});

const mapDispatchToProps = dispatch => ({
  processDeposit(amount, timestamp) {
    dispatch(deposit(amount, timestamp));
  },
  processWithdrawal(amount, timestamp) {
    dispatch(withdrawal(amount, timestamp));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(MainContainer);
