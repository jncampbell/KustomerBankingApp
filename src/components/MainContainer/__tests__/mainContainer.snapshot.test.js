import React from 'react';
import renderer from 'react-test-renderer';
import MainContainer from '../MainContainer';

const props = {
  balance: 0,
  processDeposit: jest.fn(),
  processWithdrawal: jest.fn(),
  recentTransactions: []
}

describe('MainContainer', () => {
  it('has consistent render output', () => {
    const component = renderer.create(
      <MainContainer {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
