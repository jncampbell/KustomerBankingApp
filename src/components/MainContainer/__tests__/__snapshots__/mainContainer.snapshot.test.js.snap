// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer has consistent render output 1`] = `
<div
  className="bank"
>
  <div
    className="bank-section bank__balance bank-balance"
  >
    <div
      className="bank-balance__text"
    >
      Your current balance is: 
    </div>
    <div
      className="bank-balance__amount"
    >
      $
      0.00
    </div>
  </div>
  <div
    className="bank-section"
  >
    <div
      className="bank-section__cta"
    >
      Deposit Funds
    </div>
    <div
      className="bank-section__form"
    >
      <div
        className="bank-form"
      >
        <input
          placeholder="Enter amount"
          type="number"
        />
        <button
          className="bank-form__deposit-button"
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </div>
  </div>
  <div
    className="bank-section"
  >
    <div
      className="bank-section__cta"
    >
      Withdraw Funds
    </div>
    <div
      className="bank-section__form"
    >
      <div
        className="bank-form"
      >
        <input
          placeholder="Enter amount"
          type="number"
        />
        <button
          className="bank-form__withdraw-button"
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </div>
  </div>
  <div
    className="bank-section"
  >
    <div
      className="bank-section__cta"
    >
      Show Recent Transactions
    </div>
    <div
      className="bank-recent-transactions"
    >
      <div
        className="divTable"
      >
        <div
          className="divTableBody"
        >
          <div
            className="divTableRow"
          >
            <div
              className="divTableRow"
            >
              <div
                className="divTableCell"
              >
                Date
              </div>
              <div
                className="divTableCell"
              >
                Amount
              </div>
              <div
                className="divTableCell"
              >
                Balance
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
